### DISCUSSION TITLE
Unable to load DLL 'e_sqlite3': The specified module could not be found.

### DISCUSSION THREAD

**Discussion by kmostofi** [Mimetis/Dotmim.Sync#815] [Q&A]
Hi,

I added Dotmim to a new project targeting .Net Framework 4.8.1. When I run the application, I get the following error message:

**Unable to load DLL 'e_sqlite3': The specified module could not be found.**

I have searched the net, update the other Nuget packages or downgrade them, each time a new error message popsup. After spending the whole day, I don't know what I else I can do.

Can you please advice what is missing?

Kind regards,
Kamal

**Comment by Mimetis** [comment]
Try to create a simple application using `Microsoft.Data.Sqlite`, targetting .Net `4.8.1`, and see if it's working or not.

DMS is using the version `6.0.1`

I've just tried on my side, and the result is the same as you:

![image](https://user-images.githubusercontent.com/4592555/189929848-154e7c80-9df6-4791-b2d6-47afc68ce7a6.png)

I guess `Microsoft.Data.Sqlite` is not compatible anymore with the classic `.Net Framework`


I tried to follow the steps describe here, but no chance, it does not work for me : https://github.com/dotnet/efcore/issues/19396

**Reply by kmostofi** [reply]
I tried it with a simple project and got the same error. 

I have given a presentation to my manager regarding synchronisation between Azure and SQLite (local db) today and he wanted to see it in action. :-(

What is the solution to this error? I read through the link that you shared and couldn't resolve it with their solution.

:-(

**Reply by Mimetis** [reply]
Try with `.Net Core 3.1` or `.Net 6`

**Reply by kmostofi** [reply]
Hi, I tried it with .Net 6, Visual studio 2022. It worked like a charm and without any error message.

**Reply by Mimetis** [reply]
Congratz, good luck for your demo ;)
Lest us know !

**Reply by kmostofi** [reply]
Thanks. Do I need to install ChangeTracking nuget too?

**Reply by Mimetis** [reply]
Depends if you want to use the [Change Tracking mechanism from SQL Server](https://docs.microsoft.com/en-us/sql/relational-databases/track-changes/about-change-tracking-sql-server?view=sql-server-ver16)
- If yes, use the `SqlSyncChangeTrackingProvider` provider (and don't forget to activate change tracking on your database) (example [here](https://dotmimsync.readthedocs.io/Overview.html#first-sync))
- If no, use the `SqlSyncProvider` instead, that does not require the change tracking to be enabled
